exercises/matrix/src/Matrix.hs

```diff
@@ -17,31 +17,31 @@ import Data.Vector (Vector)
 data Matrix a = Dummy deriving (Eq, Show)
 
 cols :: Matrix a -> Int
-cols = error "You need to implement this function."
+cols matrix = error "You need to implement this function."
 
 column :: Int -> Matrix a -> Vector a
-column = error "You need to implement this function."
+column x matrix = error "You need to implement this function."
 
 flatten :: Matrix a -> Vector a
-flatten = error "You need to implement this function."
+flatten matrix = error "You need to implement this function."
 
 fromList :: [[a]] -> Matrix a
-fromList = error "You need to implement this function."
+fromList xss = error "You need to implement this function."
 
 fromString :: Read a => String -> Matrix a
-fromString = error "You need to implement this function."
+fromString xs = error "You need to implement this function."
 
 reshape :: (Int, Int) -> Matrix a -> Matrix a
-reshape = error "You need to implement this function."
+reshape dimension matrix = error "You need to implement this function."
```

Comment:
try `dimensions` since there the `(Int, Int)` would mean there are two dimensions.